packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m

```diff
@@ -327,6 +341,83 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   return self;
 }
 
+/**
+ * Sets up the picture in picture controller and assigns the AVPictureInPictureControllerDelegate to
+ * the controller.
+ */
+- (void)setupPiPController {
+  if (@available(macOS 10.15, *)) {
+    if ([AVPictureInPictureController isPictureInPictureSupported]) {
+      self.pictureInPictureController =
+          [[AVPictureInPictureController alloc] initWithPlayerLayer:self.playerLayer];
+      [self setAutomaticallyStartsPictureInPicture:NO];
+      self.pictureInPictureController.delegate = self;
+    }
+  } else {
+    // We don't do anything here because there is no setup required below macOS 10.15.
+  }
+}
+
+- (void)setAutomaticallyStartsPictureInPicture:
+    (BOOL)canStartPictureInPictureAutomaticallyFromInline {
+  if (!self.pictureInPictureController) return;
+#if TARGET_OS_IOS
+  if (@available(iOS 14.2, *)) {
+    self.pictureInPictureController.canStartPictureInPictureAutomaticallyFromInline =
+        canStartPictureInPictureAutomaticallyFromInline;
+  }
+#endif
+}
+
+- (void)setPictureInPictureOverlaySettings:(CGRect)frame {
+  if (_player) {
+    self.playerLayer.frame = frame;
+  }
+}
+
+- (void)startOrStopPictureInPicture:(BOOL)shouldPictureInPictureStart {
+  if (@available(macOS 10.15, *)) {
+    if (![AVPictureInPictureController isPictureInPictureSupported] ||
+        self.isPictureInPictureStarted == shouldPictureInPictureStart) {
+      return;
+    }
+  } else {
+    return;
+  }
+
+  self.isPictureInPictureStarted = shouldPictureInPictureStart;
+  if (self.pictureInPictureController && self.isPictureInPictureStarted &&
+      ![self.pictureInPictureController isPictureInPictureActive]) {
+    if (_eventSink != nil) {
+      // The event is sent here to make sure that the Flutter UI can be updated as soon as possible.
+      _eventSink(@{@"event" : @"startedPictureInPicture"});
+    }
+    [self.pictureInPictureController startPictureInPicture];
+  } else if (self.pictureInPictureController && !self.isPictureInPictureStarted &&
```

Comment:
The `self.pictureInPictureController` check isn't necessary here; the last check is defined to return `NO` if this is `nil`.